code/framework/api/src/main/java/io/cattle/platform/api/resource/AbstractObjectResourceManager.java

```diff
@@ -459,7 +459,9 @@ protected boolean isValidAction(Object obj, Action action) {
         String state = ObjectUtils.toString(attributes.get(ObjectMetaDataManager.STATE_FIELD), null);
         String currentState = io.cattle.platform.object.util.ObjectUtils.getState(obj);
 
-        if (!StringUtils.isBlank(capability) && !DataAccessor.fieldStringList(obj, ObjectMetaDataManager.CAPABILITIES_FIELD).contains(capability)) {
+        if (!StringUtils.isBlank(capability) && !(ApiContext.getContext().getCapabilities(obj) != null ?
```

Comment:
The null check allows me to use it if they aren't null and if it is null it uses the one on the object. That is why.